.ci/azure-pipelines.yml

```diff
@@ -12,38 +12,129 @@ pr:
+          ArtifactSuffix: libre
+
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    steps:
+    - script: 'echo "##vso[task.setvariable variable=NODE_ENV]development"'
+      displayName: 'Set Node Variable'
+      condition: eq(variables['BuildConfiguration'], 'debug')
+
+    - script: 'echo "##vso[task.setvariable variable=TAG]$(git describe --tags)"'
+      displayName: 'Set Tag Variable'
+
+    - task: NodeTool@0
+      displayName: 'Install Node'
+      inputs:
+        versionSpec: '12.x'
+
+    - script: 'npm ci'
+      displayName: 'Install Dependencies'
+
+    - script: 'npx cordova prepare'
+      displayName: 'Prepare Cordova'
+
+    - script: 'npx cordova build android --debug'
+      displayName: 'Build Debug'
+      condition: eq(variables['BuildConfiguration'], 'debug')
+
+    - script: 'npx cordova plugin rm cordova-plugin-chromecast'
+      displayName: 'Remove Play Services'
+      condition: eq(variables['ArtifactSuffix'], 'libre')
+
+    - script: 'npx cordova build android --release'
+      displayName: 'Build Release'
+      condition: eq(variables['BuildConfiguration'], 'release')
+
+    - task: CopyFiles@2
+      displayName: 'Copy APK'
+      inputs:
+        sourceFolder: 'platforms/android/app/build/outputs/apk'
+        contents: '**/*.apk'
+        targetFolder: '$(Build.ArtifactStagingDirectory)'
+        flattenFolders: true
+
+    - task: Bash@3
+      displayName: 'Rename APK'
+      inputs:
+        targetType: 'inline'
+        script: 'mv *.apk jellyfin-android-${ARTIFACTSUFFIX}-${TAG}.apk'
+        workingDirectory: '$(Build.ArtifactStagingDirectory)'
+
+    - task: PublishBuildArtifacts@1
+      displayName: 'Publish APK'
+      inputs:
+        pathToPublish: '$(Build.ArtifactStagingDirectory)'
+        artifactName: 'jellyfin-android-$(ArtifactSuffix)'
+        publishLocation: 'Container'
+
+  - job: Publish
+    displayName: 'Publish'
+
+    dependsOn: Build
+    condition: startsWith(variables['Build.SourceBranch'], 'refs/tags')
+
+    strategy:
+      matrix:
+        Debug:
+          BuildConfiguration: debug
+          ArtifactSuffix: debug
+        Release:
+          BuildConfiguration: release
+          ArtifactSuffix: release
+        Libre:
+          BuildConfiguration: release
+          ArtifactSuffix: libre
+
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    steps:
+    - script: 'echo "##vso[task.setvariable variable=TAG]$(git describe --tags)"'
+      displayName: 'Set Tag Variable'
+
+    - task: DownloadPipelineArtifact@2
+      displayName: 'Download APK'
+      inputs:
+        source: 'current'
+        artifact: 'jellyfin-android-$(ArtifactSuffix)'
+        path: '$(System.ArtifactsDirectory)'
+        runVersion: 'latest'
+
+    - task: AndroidSigning@3
+      displayName: 'Sign APK'
+      condition: eq(variables['BuildConfiguration'], 'release')
+      inputs:
+        apkFiles: '$(System.ArtifactsDirectory)/*.apk'
+        keystoreFile: keystore
```

Comment:
Which keystore are we using here, just the one from build1.jellyfin.org?

Reply:
Not sure yet, the one we upload into a secure file. I mean I suppose we can have separate keys for security reasons if Google will issue us a bunch.

Reply:
OK I assumed it was already there - let me know how and I can get it uploaded securely (or @dkanada can since he also has SSH to that machine).

Reply:
I'm not sure how, probably somewhere in the settings https://docs.microsoft.com/en-us/azure/devops/pipelines/library/secure-files?view=azure-devops

Reply:
I already added this one, and you're correct.